cmd/cli/kubectl-kyverno/commands/test/test.go

```diff
@@ -98,6 +98,51 @@ func runTest(out io.Writer, testCase test.TestCase, registryAccess bool, auditWa
 	} else {
 		fmt.Fprintln(out, "  Applying", len(policies)+len(validatingAdmissionPolicies), pluralize.Pluralize(len(policies)+len(validatingAdmissionPolicies), "policy", "policies"), "to", len(uniques), pluralize.Pluralize(len(uniques), "resource", "resources"), "...")
 	}
+	rulesmap := make(map[string]string)
+	policymap := make(map[string]string)
+	for _, res := range testCase.Test.Results {
+		if res.IsValidatingAdmissionPolicy {
+			continue
+		}
+		var ns string
+		var n bool
+		r := false
+		for _, policy := range policies {
+			if policy.IsNamespaced() {
+				ns = policy.GetNamespace()
+			}
+			if res.Policy == policy.GetName() || res.Policy == "default/"+policy.GetName() || res.Policy == ns+"/"+policy.GetName() {
+				n = true
+				for _, rule := range autogen.ComputeRules(policy) {
+					if res.Rule == rule.Name {
+						r = true
+						continue
+					}
+				}
+			}
+		}
+		if !r {
+			rulesmap[res.Policy] = res.Rule
+		}
+		if !n {
+			policymap[res.Policy] = res.Policy
+		}
+	}
+	var s string
+	if len(policymap) != 0 || len(rulesmap) != 0 {
```

Comment:
No need for this condition.

Reply:
If we specify the condition for rules and policy separately, then we need to return 2 statements . so I added the outer loop which is correct then it will check the two inner loops and return statement will be in outer loop.